### ISSUE TITLE
Code for drawing the loss_landscape

### ISSUE THREAD

**Issue by hongxin001** [chenpf1025/SLN#1]
Hi, Thanks for your contribution to the label noise community.  I'm curious about how to draw the loss_landscape as you reported. Could you share the code here?

**Comment by chenpf1025** [comment]
> Hi, Thanks for your contribution to the label noise community. I'm curious about how to draw the loss_landscape as you reported. Could you share the code here?

Thanks for your interest. Sure. As cited in the notes of Figure 2, we use the visualization tool proposed by Li et al. (2018). FYI, you may be interested in their official implementation (https://github.com/tomgoldstein/loss-landscape).

**Comment by hongxin001** [comment]
Thanks for the prompt reply. I have tried the library _loss-landscapes_ for PyTorch, but it seems difficult to replicate the curve like yours. I think the problem might be due to the difference in parameters, especially the maximum distance from the start point. Could you share the parameter you used to draw the plane? It would be a great help for me.

**Comment by chenpf1025** [comment]
> Thanks for the prompt reply. I have tried the library _loss-landscapes_ for PyTorch, but it seems difficult to replicate the curve like yours. I think the problem might be due to the difference in parameters, especially the maximum distance from the start point. Could you share the parameter you used to draw the plane? It would be a great help for me.

In Li et al. (2018), they normalize the direction vector, which is very important to compare the loss landscapes on the same scale. In their official code, they have implemented the normalization by default. The only hyperparameter is the distance from the center point, for which I use 0.1, this is shown in our Figure 2.  The suitable value of distance may be dependent on network architectures and I suggest you start from a large one such as 1 then gradually tune it. What's important is that we should compare all models using the same distance.

**Comment by hongxin001** [comment]
Thanks for your kind explanation! It really helps me a lot.
